src/transformation/utils/function-context.ts

```diff
@@ -20,11 +20,23 @@ function hasNoSelfAncestor(declaration: ts.Declaration): boolean {
     if (!scopeDeclaration) {
         return false;
     } else if (ts.isSourceFile(scopeDeclaration)) {
+        if (
+            options.forceNoSelf !== undefined &&
+            typeof options.forceNoSelf === "object" &&
```

Comment:
I don't think this is necessary when you use `Array.isArray` afterwards anyway

Reply:
Sounds fair, yeah. I did some quick testing in a JS compiler online and `Array.isArray` indeed seems to have checks for undefined and the object type built in.

Good catch.